core/sys/darwin/CoreFoundation/CFString.odin

```diff
@@ -0,0 +1,77 @@
+package CoreFoundation
+
+import "core:runtime"
+
+foreign import CoreFoundation "system:CoreFoundation.framework"
+
+String :: distinct TypeRef // same as CFStringRef
```

Comment:
It looks like Apple dropped the `CFStringRef` name in Swift in favor of `CFString`. There isn't a separate `CFString` type in C, just `CFStringRef`, so seems like a safe way to go:
<img width="630" alt="Screenshot 2024-04-05 at 3 33 50 PM" src="https://github.com/odin-lang/Odin/assets/1799756/df69f900-2130-48ee-8f6e-aa1e0992c207">